src/mlpack/methods/adaboost/adaboost_fit_main.cpp

```diff
@@ -0,0 +1,180 @@
+/**
+ * @file methods/adaboost/adaboost_fit_main.cpp
+ * @author Udit Saxena
+ *
+ * Implementation of the AdaBoost main program.
+ *
+ * @code
+ * @article{Schapire:1999:IBA:337859.337870,
+ *   author = {Schapire, Robert E. and Singer, Yoram},
+ *   title = {Improved Boosting Algorithms Using Confidence-rated Predictions},
+ *   journal = {Machine Learning},
+ *   issue_date = {Dec. 1999},
+ *   volume = {37},
+ *   number = {3},
+ *   month = dec,
+ *   year = {1999},
+ *   issn = {0885-6125},
+ *   pages = {297--336},
+ *   numpages = {40},
+ *   url = {http://dx.doi.org/10.1023/A:1007614523901},
+ *   doi = {10.1023/A:1007614523901},
+ *   acmid = {337870},
+ *   publisher = {Kluwer Academic Publishers},
+ *   address = {Hingham, MA, USA},
+ *   keywords = {boosting algorithms, decision trees, multiclass classification,
+ *   output coding}
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/util/io.hpp>
+
+#ifdef BINDING_NAME
+  #undef BINDING_NAME
+#endif
+#define BINDING_NAME adaboost_fit
+
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/core/data/normalize_labels.hpp>
+#include "adaboost.hpp"
+#include "adaboost_model.hpp"
+
+using namespace mlpack;
+using namespace std;
+using namespace arma;
+using namespace mlpack::adaboost;
+using namespace mlpack::tree;
+using namespace mlpack::perceptron;
+using namespace mlpack::util;
+
+// Program Name.
+BINDING_USER_NAME("AdaBoost Training");
+
+// Short description.
+BINDING_SHORT_DESC(
+    "Training AdaBoost model.");
+
+// Long description.
+BINDING_LONG_DESC("");
+
+// Example.
+BINDING_EXAMPLE(
```

Comment:
Do you think it's worth adding a little bit of extra text here to describe what the example is doing?  Or maybe we should print comments in the generated code (e.g. inside of `CALL_METHOD()` or something)?  Or, perhaps, it's self-explanatory enough as-is?  I'm not sure, I'm just tossing ideas out there. :smile:

Reply:
We can definitely add a small text line here, but for python, I think that the example is pretty straightforward and self-explanatory. Do you have something specific in mind that you wish should be added?

Reply:
Yeah, you are right.  Let's perhaps revisit this another time, if there ever arises an example that's non-trivial. :+1: